src/main/java/com/epam/training/auction_backend/model/ZooKeeperBiddingStore.java

```diff
@@ -0,0 +1,151 @@
+import org.apache.zookeeper.ZooKeeper;
+import org.apache.zookeeper.data.Stat;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.util.SerializationUtils;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.List;
+import java.util.StringTokenizer;
+
+import static org.apache.zookeeper.CreateMode.PERSISTENT;
+import static org.apache.zookeeper.Watcher.Event.EventType.None;
+import static org.apache.zookeeper.ZooDefs.Ids.OPEN_ACL_UNSAFE;
+
+/**
+ * Created by Dmytro_Ulanovych on 10/29/2015.
+ */
+@Component
+public class ZooKeeperBiddingStore implements BiddingStore, Watcher, Closeable {
+    private static final Logger LOG = Logger.getLogger(ZooKeeperBiddingStore.class);
+    //TODO: use path from properties
+    private static final String ZOO_KEEPER_CONNECTION_PATH = "10.30.0.99:2181";
+    private static final int ZOO_KEEPER_SESSION_TTL = 2 * 60 * 1000;
+    private static final String AUCTION_BIDS_PATH = "/auction/%d/bid";
+    private ZooKeeper zooKeeper;
+    private UsersService usersService;
+
+    @Autowired
+    public ZooKeeperBiddingStore(UsersService usersService) {
+        this.usersService = usersService;
+    }
+
+    @Override
+    public UserBidTransferObject getMaxBid(long auctionId) {
+        return getHighestBidByPath(String.format(AUCTION_BIDS_PATH, auctionId));
+    }
+
+    private UserBidTransferObject getHighestBidByPath(String auctionPath) {
+        try {
+            long maxBid = getMaxBidByAuctionPath(auctionPath);
+            long bidderId = getBidderIdByPath(auctionPath + "/" + maxBid);
+            UserTransferObject user = usersService.getUserById(bidderId).get();
+            return new UserBidTransferObject(user, maxBid);
+        } catch (KeeperException | InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private long getMaxBidByAuctionPath(String auctionPath) throws KeeperException, InterruptedException {
+        List<String> bids = getZooKeeper().getChildren(auctionPath, true);
+        return bids.stream().mapToLong(Long::valueOf).max().orElse(0);
+    }
+
+    private long getBidderIdByPath(String bidPath) throws KeeperException, InterruptedException {
+        Stat stat = getZooKeeper().exists(bidPath, true);
+        byte[] serializedBidderId = getZooKeeper().getData(bidPath, true, stat);
+        return (long) SerializationUtils.deserialize(serializedBidderId);
+    }
+
+    @Override
+    public boolean makeBid(long auctionId, long amount, long userId) throws BiddingRaceException {
+        String pathForCurrentAuctionsBid = String.format(AUCTION_BIDS_PATH, auctionId);
+        createNodeIfNotExist(pathForCurrentAuctionsBid);
+        makeBid(pathForCurrentAuctionsBid, amount, userId);
+        return true;
+    }
+
+    private void makeBid(String bidsRoot, long bidValue, long userId) throws BiddingRaceException {
```

Comment:
Why do we need here this overriden private ethod ? It uses only in one method and do the logic of that public 'makeBid' method.